uber/templates/preregistration/stripeForm.html

```diff
@@ -1,16 +1,161 @@
-<form class="stripe" method="post" action="{{ action }}">
-    <input type="hidden" name="payment_id" value="{{ payment_id }}" />
-    {{ csrf_token() }}
-    <script
-        src="https://checkout.stripe.com/v2/checkout.js" class="stripe-button"
-        data-key="{{ c.STRIPE_PUBLIC_KEY }}"
-        {% if email %} data-email="{{ email }}"{% endif %}
-        {% if c.DEV_BOX %} data-bitcoin="true" {% endif %}
-        data-zip-code="true"
-        data-allow-remember-me="false"
-        data-amount="{{ charge.amount }}"
-        data-name="{{ c.EVENT_NAME_AND_YEAR }} {{ regtext }}"
-        data-description="{{ charge.description }}"
-        data-image="../static/theme/stripe-logo.png">
-    </script>
-</form>
+
+<style>
+.StripeElement {
+  box-sizing: border-box;
+
+  height: 40px;
+
+  padding: 10px 12px;
+
+  border: 1px solid transparent;
+  border-radius: 4px;
+  background-color: white;
+
+  box-shadow: 0 1px 3px 0 #e6ebf1;
+  -webkit-transition: box-shadow 150ms ease;
+  transition: box-shadow 150ms ease;
+}
+
+.StripeElement--focus {
+  box-shadow: 0 1px 3px 0 #cfd7df;
+}
+
+.StripeElement--invalid {
+  border-color: #fa755a;
+}
+
+.StripeElement--webkit-autofill {
+  background-color: #fefde5 !important;
+}
+</style>
+<button class="btn btn-info" onClick="callStripeAction();">
+<span class="display: block; min-height: 30px;">Pay with Card</span>
+</button>
+<div id="stripeModal" class="modal" style="color:black;" tabindex="-1" role="dialog" data-backdrop="static">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+        <div class="modal-header">
+        <h4 class="modal-title text-center"></h4>
+        </div>
+        <div class="modal-body text-left jumbotron">
+        <div class="charge_desc" style="margin-bottom:10px;"></div>
+        <form action="/charge" method="post" id="payment-form" class="hidden">
+        <div class="form-row">
+            <label for="card-element">
+            Credit or debit card
+            </label>
+            <div id="card-element">
+            <!-- A Stripe Element will be inserted here. -->
+            </div>
+
+            <!-- Used to display form errors. -->
+            <div id="card-errors" role="alert"></div>
+        </div>
+        <div style="margin-top:10px;"><button class="btn btn-success">Submit Payment</button>
+        <button class="btn btn-default" data-dismiss="modal">Cancel</button></div>
+        </form>
+        </div>
+    </div>
+  </div>
+</div>
+<script type="text/javascript">
+stripe_action_params = {id: '{{ id }}'};
```

Comment:
This is defined outside the function (and is a global variable) so that other payments can inject even more parameters to send to the Stripe AJAX function. In particular, the art show payment page pulls from an input form to know how much to charge a card.